app/experimenter/experiments/serializers.py

```diff
@@ -616,36 +714,65 @@ def update(self, instance, validated_data):
         return instance
 
 
-class PrefValidationMixin(object):
+class ExperimentDesignMultiPrefSerializer(ExperimentDesignBaseSerializer):
+    type = serializers.CharField()
+    is_multi_pref = serializers.BooleanField()
+    variants = ExperimentDesignBranchMultiPrefSerializer(many=True)
 
-    def validate_pref(self, pref_type, pref_value, value):
-        if pref_type == "integer":
-            try:
-                int(pref_value)
-            except ValueError:
-                return {value: "The pref value must be an integer."}
+    class Meta:
+        model = Experiment
+        fields = ("type", "is_multi_pref", "variants")
 
-        if pref_type == "boolean":
-            if pref_value not in ["true", "false"]:
-                return {value: "The pref value must be a boolean."}
+    def update(self, instance, validated_data):
+        variant_preferences = [
+            (v_d, v_d.pop("preferences")) for v_d in validated_data["variants"]
+        ]
 
-        if pref_type == "json string":
-            try:
-                json.loads(pref_value)
-            except ValueError:
-                return {value: "The pref value must be valid JSON."}
-        return {}
+        instance = super().update(instance, validated_data)
+        existing_pref_ids = self.get_existing_preference_ids(instance)
+        submitted_pref_ids = []
+        for variant_data, pref in variant_preferences:
+
+            variant = ExperimentVariant.objects.get(**variant_data)
+            for preference in pref:
+                preference["variant_id"] = variant.id
+                VariantPreferences(**preference).save()
+
+                if preference.get("id"):
+                    submitted_pref_ids.append(preference.get("id"))
+
+        removed_ids = set(existing_pref_ids) - set(submitted_pref_ids)
+
+        if removed_ids:
+            VariantPreferences.objects.filter(id__in=removed_ids).delete()
+
+        return instance
+
+    def get_existing_preference_ids(self, instance):
+        pref_ids = []
+
+        for variant in instance.variants.all():
+            pref_ids.extend([p.id for p in variant.preferences.all()])
+        return pref_ids
```

Comment:
I think you can just do

    experiment.variants.all().values_list("preferences__id", flat=True)